### PR TITLE
working on result page and django filter

### PR DESCRIPTION
**Pull request by Abdulhafeez012**

### CODE REVIEW

File: main/views.py [hunk 1]
```diff
@@ -43,29 +46,26 @@ def post(self, request, *args, **kwargs):
...
-class LogInView(View):
+class LogInView(TemplateView):
     Form = AuthenticationForm
-    template1 = 'login.html'
+    template_name = 'login.html'
 
     def get(self, request, *args, **kwargs):
         form = self.Form()
-        return render(request, self.template1, {'form': form})
+        return render(request, self.template_name, {'form': form})
```
Comment (LujainBattikhi) [inline main/views.py:55]: why did you need to use function get when using template view?

File: main/urls.py [hunk 2]
```diff
@@ -10,4 +10,5 @@
     path('register/',views.SignUp.as_view(),name='register'),
     path('home/SugJob/',views.SuggestionJobView.as_view(),name='suggestion_job'),
     path('result/',views.ResultView.as_view(),name='result'),
+    path('home/SugJob/',views.AddJob.as_view(), name='add_job'),
```
Comment (hudajundi) [inline main/urls.py:13]: the path naming could be more understandable, instead of  "SugJob" we can use " suggestions" or "suggestedJobs"

File: main/views.py (hunk 1 above)
Comment (Abdulhafeez012) [reply main/views.py:55]: i will use the get_context_data

File: main/urls.py (hunk 2 above)
Comment (Abdulhafeez012) [reply main/urls.py:13]: ok ..

File: main/urls.py [hunk 3]
```diff
@@ -5,9 +5,9 @@
 urlpatterns = [
     path('',views.BaseView.as_view(), name='base'),
     path('home/',views.HomeView.as_view(), name='home'),
-    path('logIn/',views.LogInView.as_view(), name='login'),
+    path('logIn/',views.LogInView.as_view(), name='log_in'),
     path('logout/',views.log_out, name='logout'),
     path('register/',views.SignUp.as_view(),name='register'),
-    path('home/SugJob/',views.SuggestionJobView.as_view(),name='SuggestionJob'),
+    path('home/SugJob/',views.SuggestionJobView.as_view(),name='suggestion_job'),
```
[the remark this answers is not in the document]
Comment (Abdulhafeez012) [reply main/urls.py:11]: Ok i will fixing all naming and what's must we fix on new branch

File: main/views.py [hunk 4]
```diff
@@ -75,39 +75,52 @@ def log_out(request):
...
+            'job_list' : job_list,
             }
         return render(request,self.template_name,context)
 
+    def post(self,request,*args,**kwargs):
+        user_id= UserInformation.objects.get(user=request.user.id)
+        job_id = Job.objects.get(id=request.POST.get('job_id'))
+
+        applied_job = AppliedJob.objects.create(user=user_id,
+                    job=job_id)
```
Comment (alaalqadi) [inline main/views.py:97]: it will be better to write it down as the following:
applied_job = AppliedJob.objects.create(
      user=user_id,
      job=job_id
)
this will make it readable and easy to spot maintain

File: main/views.py [hunk 5]
```diff
@@ -75,39 +75,52 @@ def log_out(request):
...
         context = {
-        'SearchBar' : SearchBar,
-        'jobs' : jobs,
+            'filter' : filter,
+            'job_model' : model,
         }
-        return render(request, self.template,context)
\ No newline at end of file
+        if request.POST.get('job_id'):
+            user_id= UserInformation.objects.get(user=request.user.id)
```
Comment (alaalqadi) [inline main/views.py:120]: user_id= UserInformation.objects.get(user=request.user.id,
to eliminate an extra join statement use 
UserInformation.objects.get(user=request.user_id)

File: requirement.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+django=4.0.1
+django_crispy_forms
+django-filter
```
Comment (alaalqadi) [inline requirement.txt:3]: to avoid compatibilities issues in the future always define the version number after the name, like django=x.x.x

File: templates/base.html [hunk 7]
```diff
@@ -13,73 +13,35 @@
         integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p"
         crossorigin="anonymous"></script>
     <link href="{% static 'css/base.css' %}" rel="stylesheet">
-    <title>wasefah</title>
+    <link href="{% static 'css/result_page.css' %}" rel="stylesheet">
+    <title>Wazefah</title>
 </head>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js"
+    integrity="sha512-894YE6QWD5I59HgZOGReFYm4dnWc1Qt5NtvYSaNcOP+u1T9qYdvdihz0PPSiiqn/+/3e7Jo4EaG7TubfWGUrMQ=="
+    crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
Comment (alaalqadi) [inline templates/base.html:21]: add it as a local package, not CDN

File: templates/base.html [hunk 8]
```diff
@@ -13,73 +13,35 @@
...
+            {% for message in messages %}
+            <div class="alert alert-danger d-flex justify-content-center mx-5" role="alert" style="margin-top: 150px;">
+                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="currentColor"
+                    class="bi bi-exclamation-triangle-fill flex-shrink-0 me-2" viewBox="0 0 16 16" role="img"
+                    aria-label="Warning:">
+                    <path
+                        d="M8.982 1.566a1.13 1.13 0 0 0-1.96 0L.165 13.233c-.457.778.091 1.767.98 1.767h13.713c.889 0 1.438-.99.98-1.767L8.982 1.566zM8 5c.535 0 .954.462.9.995l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 5.995A.905.905 0 0 1 8 5zm.002 6a1 1 0 1 1 0 2 1 1 0 0 1 0-2z" />
+                </svg>
+                <div>
+                    {{ message }}
```
Comment (alaalqadi) [inline templates/base.html:38]: try to display the message alert color based on the message type

**Review by alaalqadi — CHANGES_REQUESTED** [review]
